extensions/service_bindings_manager.go

```diff
@@ -0,0 +1,76 @@
+package extensions
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// ServiceBinding is a representation of a service binding that has
+// been provided into the build container.
+type ServiceBinding struct {
+	// Name is the name of the directory that binding was contained within.
+	Name string
+
+	// Type is the content of the type file contained within the binding
+	// directory.
+	Type string
+
+	// Provider is the content of the provider file contained within the binding
+	// directory.
+	Provider string
+
+	// Path is the filepath of the binding directory.
+	Path string
+
+	// Secrets is a mapping of file name to contents for all other files
+	// contained in the bindings directory.
+	Secrets map[string][]byte
+}
+
+// ListServiceBindings returns all service bindings found in the given root
```

Comment:
This could be premature, but could it be useful to provide a mechanism to get a single binding (where you know what binding name you're looking for)? That way you don't have to load all the bindings every time you need only a specific one. Since the bindings will presumably be in place for all the buildpacks, I could see a scenario where one buildpack requires a large number of bindings, while another buildpack in the group only needs one or two.

Reply:
Is your idea here about improving UX for packit consumers or improving build performance/speed?